Tasks/task1 completed by Юрченко Кирило

UnoMomento12 opened the pull request.

KyryloYurchenko committed:
1st commit

KyryloYurchenko committed:
2nd commit

KyryloYurchenko committed:
Menu is implemented

KyryloYurchenko committed:
Third commit

Added comments and 2 bug fixes

KyryloYurchenko committed:
Moved project files to the folder

UnoMomento12 committed:
Update TextHolder.cs

UnoMomento12 committed:
Update TextHolder.cs

Minor bug fix.

KyryloYurchenko committed:
Added gitignore

KyryloYurchenko committed:
Updated gitignore

KyryloYurchenko committed:
And another update of gitignore

KyryloYurchenko committed:
commit in passing

KyryloYurchenko committed:
Merge remote-tracking branch 'origin/tasks/task1' into tasks/task1

KyryloYurchenko committed:
Deleting unnecessary folders

KyryloYurchenko committed:
Enhancing SOLID principles

KyryloYurchenko committed:
I hope it's the last one.

alex-geraschenko commented on task1/CustomDirectorySelector.cs:
```diff
@@ -0,0 +1,95 @@
+            {
+                if (directoryNames.Count == 0 || directoryNames.Count <= id || id < -1 ) return;
+                dirPath = dirPath + "\\" + directoryNames[id];
+                directoryNames.Clear();
```
Постарайтесь уйти от дублирования кода

alex-geraschenko commented on task1/CustomFileHandler.cs:
```diff
@@ -0,0 +1,68 @@
+            string currentLine = null;
+            try
+            {
+                using (StreamReader sr = new StreamReader(filePathToRead, Encoding.Default))
```
Обратите внимание на статические методы из `File`

alex-geraschenko commented on task1/FileTaskExecutors/SecondTaskFileExecutor.cs:
```diff
@@ -0,0 +1,48 @@
+
+        public override void Execute(params string[] args)
+        {
+            string filePath = !String.IsNullOrEmpty(args[0]) ? args[0].Replace("\"", "") : "";// Deletes quotes in the string
```
Постарайтей избавиться от дублирования кода

alex-geraschenko commented on task1/TextHolder.cs:
```diff
@@ -0,0 +1,113 @@
+        /// <summary>
+        /// An array of lines in the text.
+        /// </summary>
+        public List<string> lines;
```
1. Вместо пибличных полей лучше использовать публичные свойства
2. Публичные свойства принято называть в `PascalCase` нотации.

alex-geraschenko commented on task1/CustomFileHandler.cs:
```diff
@@ -0,0 +1,68 @@
+                    while ((currentLine = sr.ReadLine()) != null)
+                    {
+                        //if (!String.IsNullOrEmpty(currentLine))
+                            textHolder.lines.Add(currentLine);
```
Проще работать с цельным текстом, а не разбивать его на строки

KyryloYurchenko committed:
Changes applied

1. Текст из файла считывается в одну string'у сразу  и вся работа в дальнейшем идет с ней.
2. Сменил паблик поля на свойства.
3. Убрал повторения кода и ненужные повторные null checks.
4. Запись в файл также теперь идет с одной стринги.

KyryloYurchenko committed:
Deleted another code repeat

alex-geraschenko commented on task1/FileTaskExecutors/SecondTaskFileExecutor.cs:
```diff
@@ -0,0 +1,48 @@
+
+                if (File.Exists(filePath)) // If path directs to existing file, execute following:
+                {
+                    FileInfo fileInfo = new FileInfo(filePath);
```
Зачем нужен `fileInfo` ?

alex-geraschenko commented on task1/FileTaskExecutors/ThirdTaskFileExecutor.cs:
```diff
@@ -0,0 +1,51 @@
+
+                if (File.Exists(filePath)) // If path directs to existing file, execute following:
+                {
+                    FileInfo fileInfo = new FileInfo(filePath);
```
Зачем нужен `fileInfo` ?

alex-geraschenko commented on task1/FileTaskExecutors/FirstTaskFileExecutor.cs:
```diff
@@ -0,0 +1,56 @@
+        {
+            string filePath = CustomFileHandler.SetStringOrDefault(args[0], "");
+            string wordToDelete = CustomFileHandler.SetStringOrDefault(args[1], "");
+            filePath = filePath.Replace("\"", ""); // Deletes quotes in the string
```
Подготовка `filePath` дублируется. Нужно вынейсти в общий код

alex-geraschenko commented on task1/FileTaskExecutors/SecondTaskFileExecutor.cs:
```diff
@@ -0,0 +1,48 @@
+        public override void Execute(params string[] args)
+        {
+            string filePath = CustomFileHandler.SetStringOrDefault(args[0], ""); // Deletes quotes in the string
+            filePath = filePath.Replace("\"", ""); // Deletes quotes in the string
```
Подготовка `filePath` дублируется. Нужно вынейсти в общий код

alex-geraschenko commented on task1/FileTaskExecutors/ThirdTaskFileExecutor.cs:
```diff
@@ -0,0 +1,51 @@
+        public override void Execute(params string[] args)
+        {
+            string filePath = CustomFileHandler.SetStringOrDefault(args[0], "");
+            filePath = filePath.Replace("\"", ""); // Deletes quotes in the string
```
Подготовка `filePath` дублируется. Нужно вынейсти в общий код

alex-geraschenko commented on task1/FileTaskExecutors/ThirdTaskFileExecutor.cs:
```diff
@@ -0,0 +1,51 @@
+            while (true)
+            {
+
+                if (File.Exists(filePath)) // If path directs to existing file, execute following:
```
Проверка на существование файла дублируется. Нужно вынести в общий код

alex-geraschenko commented on task1/FileTaskExecutors/SecondTaskFileExecutor.cs:
```diff
@@ -0,0 +1,48 @@
+            while (true)
+            {
+
+                if (File.Exists(filePath)) // If path directs to existing file, execute following:
```
Проверка на существование файла дублируется. Нужно вынести в общий код

alex-geraschenko commented on task1/FileTaskExecutors/FirstTaskFileExecutor.cs:
```diff
@@ -0,0 +1,56 @@
+            cstTextHolder.ClearTextHolder();
+            while (true)
+            {
+                if (File.Exists(filePath)) // If path directs to existing file, execute following:
```
Проверка на существование файла дублируется. Нужно вынести в общий код

alex-geraschenko commented on task1/FileTaskExecutors/FirstTaskFileExecutor.cs:
```diff
@@ -0,0 +1,56 @@
+
+                    CustomFileHandler.SaveOriginalFile(filePath);// save original file with .orig suffix
+                    
+                    cstFile.ReadTextFromFile(filePath); // Read lines of text from file
```
Чтение из файла дублируется. Нужно вынести в общий код

alex-geraschenko commented on task1/FileTaskExecutors/SecondTaskFileExecutor.cs:
```diff
@@ -0,0 +1,48 @@
+                {
+                    FileInfo fileInfo = new FileInfo(filePath);
+
+                    cstFile.ReadTextFromFile(filePath);
```
Чтение из файла дублируется. Нужно вынести в общий код

alex-geraschenko commented on task1/FileTaskExecutors/ThirdTaskFileExecutor.cs:
```diff
@@ -0,0 +1,51 @@
+                if (File.Exists(filePath)) // If path directs to existing file, execute following:
+                {
+                    FileInfo fileInfo = new FileInfo(filePath);
+                    cstFile.ReadTextFromFile(filePath); // Read lines from file
```
Чтение из файла дублируется. Нужно вынести в общий код

alex-geraschenko commented on task1/Program.cs:
```diff
@@ -0,0 +1,102 @@
+            switch (args[0])
+            {
+                case "func1":
+                    path = args.Length > 1 ? args[1] : null;
```
Дублирование кода

alex-geraschenko commented on task1/Program.cs:
```diff
@@ -0,0 +1,102 @@
+                    taskExecutor = new FirstTaskFileExecutor(txtHolder);
+                    break;
+                case "func2":
+                    path = args.Length > 1 ? args[1] : "";
```
Дублирование кода

alex-geraschenko commented on task1/Program.cs:
```diff
@@ -0,0 +1,102 @@
+                    
+                    break;
+                case "func3":
+                    path = args.Length > 1 ? args[1] : "";
```
Дублирование кода

alex-geraschenko commented on task1/Program.cs:
```diff
@@ -0,0 +1,102 @@
+                    taskExecutor = new ThirdTaskFileExecutor(txtHolder);
+                    break;
+                case "func4":
+                    path = args.Length > 1 ? args[1] : "";
```
Дублирование кода

alex-geraschenko commented on task1/Program.cs:
```diff
@@ -0,0 +1,102 @@
+            {
+                case "func1":
+                    path = args.Length > 1 ? args[1] : null;
+                    secondParam = args.Length > 2 ? args[2] : null;
```
Дублирование кода

alex-geraschenko commented on task1/Program.cs:
```diff
@@ -0,0 +1,102 @@
+                    break;
+                case "func3":
+                    path = args.Length > 1 ? args[1] : "";
+                    secondParam = args.Length > 2 ? args[2] : "3";
```
Дублирование кода

alex-geraschenko commented on task1/TaskInterfaces/ITaskExecutor.cs:
```diff
@@ -0,0 +1,11 @@
+{
+    interface ITaskExecutor
+    {
+        public void Execute(params string[] args) { }
```
У методов интерфейса не может быть модификатора доступа. В интерфейсе все методы публичные.

KyryloYurchenko committed:
Quite big code clean-up.

alex-geraschenko reviewed (approved):
Выглядит хорошо. Можно мержить
